app/src/main/java/com/example/appmarvel/adapter/CharacterAdapter.kt

```diff
@@ -0,0 +1,39 @@
+package com.example.appmarvel.adapter
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.Glide
+import com.example.appmarvel.R
+import com.example.appmarvel.databinding.CardViewCharacterBinding
+import com.example.appmarvel.entity.Character
+
+class CharacterAdapter(private val charactersList: List<Character>) :
+    RecyclerView.Adapter<CharacterAdapter.ViewHolder>() {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        return ViewHolder(
+            LayoutInflater.from(parent.context).inflate(R.layout.card_view_character, parent, false)
+        )
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        holder.bind(charactersList[position])
+    }
+
+    override fun getItemCount(): Int = charactersList.size
+
+    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+        private val binding = CardViewCharacterBinding.bind(itemView)
+        fun bind(item: Character) {
+            item.let {
+                binding.apply {
+                    this.textViewCharacterName.text = item.name
+                    Glide.with(itemView.context).load(item.imageURL)
```

Comment:
please format this better to grant legibility 

```
Glide.with(itemView.context)
.load(item.imageURL)
.into(this.imageViewCharacterPicture)
```

Reply:
Done